app/controllers/application_controller.rb

```diff
@@ -14,4 +15,12 @@ def after_sign_out_path_for(resource)
   def after_update_path_for(resource)
     edit_user_registration_path
   end
+
+  def set_locale
+    I18n.locale = params[:locale]
```

Comment:
rename `locale` to `l` and use it as a parameter, e.g. `otvorenesudy.sk/?l=en` not as `otvorenesudy.sk/sk`.